python/rpmmodules.c: Change the way the python module loads the RPM config

In order to support the RPM_VENDOR_WINDRIVER enhancement of dynamic
runtime relocation paths, we need to call rpmcliInit instead of 
rpmReadConfigFiles.  The rpmcliInit will end up calling rpmReadConfigFiles
after the necessary relocation processing (if enabled).

Code derived from changes suggested by Paul Eggleton.

Upstream-Status: Pending

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: rpm/python/rpmmodule.c
===================================================================
--- rpm.orig/python/rpmmodule.c
+++ rpm/python/rpmmodule.c
@@ -382,9 +382,8 @@ static int initModule(PyObject *m)
     /* XXX add --noparentdirs --nolinktos to rpmtsCheck() */
     global_depFlags = (RPMDEPS_FLAG_NOPARENTDIRS | RPMDEPS_FLAG_NOLINKTOS);
 
-    /* failure to initialize rpm (crypto and all) is rather fatal too... */
-    if (rpmReadConfigFiles(NULL, NULL) == -1)
-	return 0;
+    const char *argv[1] = {"rpmmodule", 0};
+    rpmcliInit(1, argv, NULL);
 
     d = PyModule_GetDict(m);
 
